### PR TITLE
Query History: Implement RemoteStorage methods: settings

### LINKED ISSUE #48408: Query History: Implement RemoteStorage methods: settings
Remote storage will support only setting home tab (starred or regular search). 

All settings should still be available when query history is not enabled.

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

This PR adds methods to update and load user preferences for query history. The settings can be changes in the Query History settings tab:

Screen Shot 2022-05-20 at 13 32 12

When query history is disabled it shows the old settings panel with more options. 

**Which issue(s) this PR fixes**:

Fixes #48408

**Special notes for your reviewer**:

How to test it?

Make sure query history is enabled in `custom.ini`:

```
[query_history]
enabled = true
```

Change settings to make starred tab default and check if it's selected when the container is opened (also after refreshing the page).

There's one caveat: There's a bug when settings are updated from User Preferences section. It overrides all existing preferences. `queryHistory` settings are passed to keep them intact but they are not being saved to the database. I've created an issue for it #49322.

### CODE REVIEW

File: public/app/features/explore/RichHistory/RichHistorySettingsTab.tsx [hunk 1]
```diff
@@ -73,46 +74,58 @@ export function RichHistorySettingsTab(props: RichHistorySettingsProps) {
...
+          description={`Select the period of time for which Grafana will save your query history. Up to ${MAX_HISTORY_ITEMS} entries will be stored.`}
+          className="space-between"
+        >
+          <div className={styles.input}>
+            <Select value={selectedOption} options={retentionPeriodOptions} onChange={onChangeRetentionPeriod}></Select>
+          </div>
+        </Field>
+      ) : (
+        <Alert severity="info" title="History time span">
+          Grafana will keep entries that are no older than {selectedOption?.label}.
```
Comment (gelicia) [inline public/app/features/explore/RichHistory/RichHistorySettingsTab.tsx:89]: How about "Grafana will keep entries up to 2 weeks." ?

**Review by gelicia — APPROVED** [review]

File: public/app/types/preferences.ts [hunk 2]
```diff
@@ -5,4 +5,7 @@ export interface UserPreferencesDTO {
   weekStart: string;
   homeDashboardId: number;
   theme: string;
+  queryHistory: {
+    homeTab: '' | 'query' | 'starred';
```
[the remark this answers is not in the document]
Comment (ifrost) [reply public/app/types/preferences.ts:9]: Just wanted to be more explicit. `homeTab` is always provided but it comes as an empty string from the backend if it's not defined.

[the remark this answers is not in the document]
Comment (Elfo404) [reply public/app/types/preferences.ts:9]: ah, i missed that part, if the BE returns an empty string then lgtm 🚀

File: public/app/core/history/RichHistoryRemoteStorage.ts [hunk 3]
```diff
@@ -84,7 +98,11 @@ export default class RichHistoryRemoteStorage implements RichHistoryStorage {
   }
 
   async updateSettings(settings: RichHistorySettings): Promise<void> {
-    throw new Error('not supported yet');
+    await this.preferenceService.patch({
```
[the remark this answers is not in the document]
Comment (ifrost) [reply public/app/core/history/RichHistoryRemoteStorage.ts]: Updated in 767bab2522b2d4a95ffaf076e39378225a427325

**Review by Elfo404 — APPROVED** [review]

**Pull request merged**